src/frontend/src/error.rs

```diff
@@ -254,3 +257,35 @@ impl From<BoxedError> for RwError {
         ErrorCode::Uncategorized(e).into()
     }
 }
+
+/// Result when applying an optimization rule.
+pub enum OResult<T> {
```

Comment:
Shall we move this definition to the `optimizer` module?

Also, it seems that there's no `Try` implementation?

Reply:
> Also, it seems that there's no `Try` implementation?

Oh, I realize that we need it only if we want to apply `?` to an `OResult`. Since we usually don't have nested call of `apply`, I think it's okay to omit it.